Sources/NIOHTTP1/HTTPEncoder.swift

```diff
@@ -78,12 +78,13 @@ private func writeHead(wrapOutboundOut: (IOData) -> NIOAny, writeStartLine: (ino
 
     var buffer = ctx.channel.allocator.buffer(capacity: 256)
     writeStartLine(&buffer)
-    headers.write(buffer: &buffer)
+    headers.write(into: &buffer)
     ctx.write(wrapOutboundOut(.byteBuffer(buffer)), promise: promise)
 }
 
-private func isChunkedPart(_ headers: HTTPHeaders) -> Bool {
-    return headers["transfer-encoding"].contains("chunked")
+private enum Chunked {
```

Comment:
Let's call this `enum BodyFraming` and have three cases, `.chunked`, `contentLength`, and `neither`.